service/worker/scanner/shardscanner/types.go

```diff
@@ -419,6 +414,69 @@ func (s Shards) Flatten() ([]int, int, int) {
 	return shardList, min, max
 }
 
-func (c ScannerContextKey) String() string {
-	return string(c)
+// NewShardScannerContext sets scanner context up
+func NewShardScannerContext(
+	res resource.Resource,
+	config *ScannerConfig,
+) Context {
+	return Context{
+		Resource: res,
+		Scope:    res.GetMetricsClient().Scope(metrics.ExecutionsScannerScope),
+		Config:   config,
+		Hooks:    config.ScannerHooks(),
+	}
+}
+
+// NewShardFixerContext sets fixer context up
+func NewShardFixerContext(
+	res resource.Resource,
+	config *ScannerConfig,
+) FixerContext {
+	return FixerContext{
+		Resource: res,
+		Scope:    res.GetMetricsClient().Scope(metrics.ExecutionsFixerScope),
+		Config:   config,
+		Hooks:    config.FixerHooks(),
+	}
+}
+
+// NewContext provides context to be used as background activity context
+func NewContext(
+	ctx context.Context,
+	workflowName string,
+	scannerContext interface{},
+) context.Context {
+	return context.WithValue(ctx, contextKey(workflowName), scannerContext)
+}
+
+// GetScannerContext extracts scanner context from activity context
+// it uses typed, private key to reduce access scope
+func GetScannerContext(
+	ctx context.Context,
+) (*Context, error) {
+	info := activity.GetInfo(ctx)
+	if info.WorkflowType == nil {
+		return nil, fmt.Errorf("workflowType is nil")
+	}
+	val, ok := ctx.Value(contextKey(info.WorkflowType.Name)).(*Context)
+	if !ok {
+		return nil, fmt.Errorf("context type is not %T for a key %q", val, info.WorkflowType.Name)
+	}
+	return val, nil
+}
+
+// GetFixerContext extracts fixer context from activity context
+// it uses typed, private key to reduce access scope
+func GetFixerContext(
```

Comment:
GetScannerContext and GetFixerContext are almost the same except for casting. Maybe just have a pair of symmetric methods:
```
SetActivityContext
```
and
```
GetActivityContext
```